File: 02_UML/Object_Association/Simple_Association/Main.cpp
#include <iostream>
using namespace std;

class Manager {
public:
    void manage() {
        cout << "Manager is managing\n";
    }
};

class Employee {
public:
    void workWith(Manager& manager) {
        manager.manage();
    }
};

int main() {
    Manager manager;
    Employee employee;

    employee.workWith(manager);
}

File: 02_UML/Object_Association/Simple_Association/Main.java
class Manager {
    void manage() {
        System.out.println("Manager is managing");
    }
}

class Employee {
    void workWith(Manager manager) {
        manager.manage();
    }
}

public class Main {
    public static void main(String[] args) {
        Manager manager = new Manager();
        Employee employee = new Employee();

        employee.workWith(manager);
    }
}